packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingTestState.ts

```diff
@@ -700,16 +701,43 @@ export class MappingTestState {
     this.test.setAssert(this.assertionState.assert);
   }
 
-  *generatePlan(): GeneratorFn<void> {
+  *generatePlan(debug: boolean): GeneratorFn<void> {
     try {
       this.isGeneratingPlan = true;
-      yield flowResult(
-        this.executionPlanState.generatePlan(
-          this.mappingEditorState.mapping,
-          this.queryState.query,
-          this.inputDataState.runtime,
-        ),
-      );
+      let rawPlan: RawExecutionPlan;
+      if (debug) {
+        const debugResult =
+          (yield this.editorStore.graphManagerState.graphManager.debugExecutionPlanGeneration(
+            this.editorStore.graphManagerState.graph,
+            this.mappingEditorState.mapping,
+            this.queryState.query,
+            this.inputDataState.runtime,
+            PureClientVersion.VX_X_X,
+          )) as { plan: RawExecutionPlan; debug: string };
+        rawPlan = debugResult.plan;
+        this.executionPlanState.setDebugText(debugResult.debug);
+        this.executionPlanState.setShowDebugPanel(true);
+      } else {
+        rawPlan =
```

Comment:
same comment as for service

Reply:
that's a fair point, I actually did this for `Legend Query`, I think I can skim down that.